src/SecureStore.BeyondTrust/BeyondTrustSingleSystemSecureStore.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        public Task ValidateContextAsync(string context)
+        {
+            BeyondTrustVaultClientFactory.GetClient(context).TestConnection();
+            return Task.CompletedTask;
+        }
+
+        public IEnumerable<ConfigurationEntry> GetConfiguration()
+        {
+            return new List<ConfigurationEntry>
+            {
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.Hostname,
+                    DisplayName = "BeyondTrust Host URL",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.AuthKey,
+                    DisplayName = "API Authentication Key",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.RunAs,
+                    DisplayName = "API Run As",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.Boolean)
+                {
+                    Key = ConfigurationConstants.SSLEnabled,
+                    DisplayName = "Use SSL certificate",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.String)
+                {
+                    Key = ConfigurationConstants.ManagedSystemName,
+                    DisplayName = "Managed System Name",
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.Choice)
+                {
+                    Key = ConfigurationConstants.ManagedAccountType,
+                    DisplayName = "Managed Account Type",
+                    IsMandatory = true,
+                    PossibleValues = new List<string> { ConfigurationConstants.SystemAccountType, ConfigurationConstants.DomainAccountType }
+                },
+            };
+        }
+
+        public SecureStoreInfo GetStoreInfo()
+        {
+            return new SecureStoreInfo { Identifier = "BeyondTrust-SingleSystem-ReadOnly", IsReadOnly = true };
+        }
+
+        public async Task<string> CreateCredentialsAsync(string context, string key, Credential value)
+        {
+            throw new SecureStoreException(SecureStoreException.Type.UnsupportedOperation, "This secure store is read-only");
+        }
+
+        public async Task<Credential> GetCredentialsAsync(string context, string key)
+        {
+            var config = JsonConvert.DeserializeObject<Dictionary<string, object>>(context);
+            var client = BeyondTrustVaultClientFactory.GetClient(context);
+            var semaphore = SemaphoreFactory.SemaphoreSlim;
+            await semaphore.WaitAsync();
```

Comment:
You should use a timeout, so as not to wait forever in case there is a block somewhere.
```suggestion
            await semaphore.WaitAsync(TimeSpan.FromSeconds(60));
```